MediaGalleryUi/view/adminhtml/web/js/directory/directoryTree.js

```diff
@@ -203,21 +210,33 @@ define([
          *  Handle jstree events
          */
         initEvents: function () {
+            this.firejsTreeEvents();
             this.overrideMultiselectBehavior();
 
-            $(this.directoryTreeSelector).on('loaded.jstree', function () {
-                this.checkChipFiltersState();
-            }.bind(this));
-
             $(window).on('reload.MediaGallery', function () {
-                this.checkChipFiltersState();
+                this.renderDirectoryTree().then(function () {
+                    var path = this.filterChips().get('applied');
+                    this.filterChips().clear();
+                    this.filterChips().set('applied', path);
```

Comment:
Why do we get the applied filter, clear, and apply the same data again (without changes), is that required here?

Reply:
this is like reload event for filters, to avoid double selecting folder in directory tree after reloading

Reply:
How "double selecting folder" issue can be reproduced?

Reply:
When we reload jstree, selected folder cleared from jstree, and it tries to select directory again for example when we open second time category widget with `current_tree_path="catalog/category"`, and we have already applied directory filter `catalog/category,` then it tries to select `catalog/category `per current_tree_path, but it already applied in filter so second click will remove directory filter. so we also need reload the directory filter when we reload the jstree.

Reply:
we also can only call clear() filters, but if we have predefined directory filter in ui bookmarks wich is not catalog/category we need to save this filter and apply after reloading jstree.